src/system/SystemPacketBuffer.cpp

```diff
@@ -431,6 +431,7 @@ void PacketBuffer::AddRef()
     pbuf_ref(this);
 #else  // !CHIP_SYSTEM_CONFIG_USE_LWIP
     LOCK_BUF_POOL();
+    VerifyOrDieWithMsg(this->ref < UINT16_MAX, chipSystemLayer, "packet buffer refcount overflow");
```

Comment:
Please use `std::numeric_limits<decltype(this->ref)>` so that you're not vulnerable to the underlying type of `ref` changing in the future.